src/main/java/com/serhiiklymchuk/gmailpet/service/impl/MessageServiceImpl.java

```diff
@@ -0,0 +1,84 @@
+package com.serhiiklymchuk.gmailpet.service.impl;
+
+import com.serhiiklymchuk.gmailpet.domain.Message;
+import com.serhiiklymchuk.gmailpet.domain.User;
+import com.serhiiklymchuk.gmailpet.dto.MessageDto;
+import com.serhiiklymchuk.gmailpet.dto.MessageFormDto;
+import com.serhiiklymchuk.gmailpet.repository.MessageRepository;
+import com.serhiiklymchuk.gmailpet.service.MessageService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class MessageServiceImpl implements MessageService {
+
+    private final MessageRepository messageRepository;
+
+    private final UserServiceImpl userService;
+
+    public MessageServiceImpl(MessageRepository messageRepository, UserServiceImpl userService) {
+        this.messageRepository = messageRepository;
+        this.userService = userService;
+    }
+
+    public List<MessageDto> getInboxMessages(User user) throws RuntimeException {
+
+        List<Message> messages = messageRepository
+                .findAllByReceiverUserIdOrderByDateDesc(user.getId())
+                .orElseThrow(() -> new RuntimeException("Inbox messages not found!"));
+
+        return messages.stream().map(msg -> MessageDto.builder()
```

Comment:
Remapping should be moved to separate class
`mapper.MessageMapper`
But, I don't see the need in 1:1 remapping, for JBDC entity, as long as, it's not in the session, more like a simple POJO

Reply:
https://github.com/SerhiiKlymchuk/gmail-pet/pull/10#discussion_r563147074

Reply:
The comment was about moving mapper to separate class.
Sample: https://github.com/PNUdev/radio-service/blob/master/radio-service-api/src/main/java/com/pnu/dev/radioserviceapi/util/mapper/ScheduleItemToScheduleItemDtoMapper.java#L15